### PR TITLE
`ultralytics 8.0.26` new YOLOv5u models

### PR DESCRIPTION
**Pull request by glenn-jocher**
Adds new and improved YOLOv5u models to replace YOLOv5 models. YOLOv5u models use the same loss functions and split Detect head as YOLOv8 models for improved performance. The 'u' stands for ultralytics, to denote that they were trained using the ultralytics repo so they can be differentiated from standard YOLOv5 models.

Usage example:

```bash
yolo predict model=yolov5nu.pt
```

See [Detection Docs](https://docs.ultralytics.com/tasks/detection/) for usage examples with these models.

| Model                                                                                  | size<br><sup>(pixels) | mAP<sup>val<br>50-95 | Speed<br><sup>CPU ONNX<br>(ms) | Speed<br><sup>A100 TensorRT<br>(ms) | params<br><sup>(M) | FLOPs<br><sup>(B) |
|----------------------------------------------------------------------------------------| --------------------- |----------------------|--------------------------------|-------------------------------------|--------------------|-------------------|
| [YOLOv5nu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5nu.pt) | 640                   | 34.3                 | 73.6                           | 1.06                                | 2.6                | 7.7               |
| [YOLOv5su](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5su.pt) | 640                   | 43.0                 | 120.7                          | 1.27                                | 9.1                | 24.0              |
| [YOLOv5mu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5mu.pt) | 640                   | 49.0                 | 233.9                          | 1.86                                | 25.1               | 64.2              |
| [YOLOv5lu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5lu.pt) | 640                   | 52.2                 | 408.4                          | 2.50                                | 53.2               | 135.0             |
| [YOLOv5xu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5xu.pt) | 640                   | 53.2                 | 763.2                          | 3.81                                | 97.2               | 246.4             |

- **mAP<sup>val</sup>** values are for single-model single-scale on [COCO val2017](http://cocodataset.org) dataset.
  Reproduce by `yolo val data=coco.yaml device=0`
- **Speed** averaged over COCO val images using an [Amazon EC2 P4d](https://aws.amazon.com/ec2/instance-types/p4/)
  instance.
  Reproduce by `yolo va data=coco128.yaml batch=1 device=0/cpu`

EDIT: Waiting on TensorRT results
EDIT2: Table complete! :)

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancements and clean-up in Ultralytics' software stack!

### 📊 Key Changes
- Updated the base Docker image to the latest PyTorch version.
- Removed version constraint for opencv-python in Dockerfile, allowing the latest compatible version to be installed.
- Renamed model configuration files to include a 'u' for Ultralytics specific models.
- Cleaned up Python code: simplified loops and improved error messages and type checks.
- Improved Dockerfile clarity and reduced redundancy in Dockerfile installation commands.
- Streamlined callback functions and cleaned up code comments.
- Set the execution counts in Jupyter notebooks to `null`.

### 🎯 Purpose & Impact
- Ensures compatibility with the latest PyTorch version and hardware acceleration 🚀.
- Unpins opencv-python version, potentially allowing more recent versions with important fixes.
- Model renaming provides clear differentiation between standard YOLO models and those optimized by Ultralytics 🆕.
- Code clean-up enhances maintainability and contributes to easier comprehension and debugging for developers 🛠️.
- Notebook updates may decrease confusion for users importing the tutorial to work with 📚.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
See [Detection Docs](https://docs.ultralytics.com/tasks/detection/) for usage examples with these models.

| Model                                                                                  | size<br><sup>(pixels) | mAP<sup>val<br>50-95 | Speed<br><sup>CPU ONNX<br>(ms) | Speed<br><sup>A100 TensorRT<br>(ms) | params<br><sup>(M) | FLOPs<br><sup>(B) |
|----------------------------------------------------------------------------------------| --------------------- |----------------------|--------------------------------|-------------------------------------|--------------------|-------------------|
| [YOLOv5nu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5nu.pt) | 640                   | 34.3                 | 73.6                           | -                                   | 2.6                | 7.7               |
| [YOLOv5su](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5su.pt) | 640                   | 43.0                 | 120.7                          | -                                   | 9.1                | 24.0              |
| [YOLOv5mu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5mu.pt) | 640                   | 49.0                 | 233.9                          | -                                   | 25.1               | 64.2              |
| [YOLOv5lu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5lu.pt) | 640                   | 52.2                 | 408.4                          | -                                   | 53.2               | 135.0             |
| [YOLOv5xu](https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov5xu.pt) | 640                   | 53.2                 | 763.2                          | -                                   | 97.2               | 246.4             |

- **mAP<sup>val</sup>** values are for single-model single-scale on [COCO val2017](http://cocodataset.org) dataset.
  <br>Reproduce by `yolo val data=coco.yaml device=0`
- **Speed** averaged over COCO val images using an [Amazon EC2 P4d](https://aws.amazon.com/ec2/instance-types/p4/)
  instance.
  <br>Reproduce by `yolo va data=coco128.yaml batch=1 device=0/cpu`

**Comment by glenn-jocher** [comment]
@adrianboguszewski does OpenVINO offer a simple benchmarking command? To create our official YOLOv8 tables I've been using a custom script I put together. If there were a similar OpenVINO script that would help us evaluate models there. 

```bash
# Profile ONNX
!gsutil cp gs://ultralytics/onnx_speed.py .
!pip install onnxruntime
from pathlib import Path
for f in sorted(Path().glob('*.onnx')):
  print(f'\n\nSTARTING PROFILING FOR {f} -----------------------------------------------------------------------------')
  !python onnx_speed.py {f} 100 --profile
```

**Comment by glenn-jocher** [comment]
@adrianboguszewski thanks! I've opened up #777 to track OpenVINO benchmarking efforts separately.
